### ISSUE TITLE
Support config at spec, container and test level.

### ISSUE THREAD

**Issue by crazyk2** [kotest/kotest#1370]
As I describe there #1369 ContextScope doesn't allow you use `threads` and `invocations` parameter.
But you can define default config via several ways
```kotlin
class FunSpecContextConfigTest : FunSpec({

   defaultTestConfig = TestCaseConfig(invocations = 5, threads = 2)

   test("a") {
      println("a test")

   }
   context("foo") {
      test("a") {
         println("foo a test")
      }
   }
}) {
   override fun defaultTestCaseConfig() = TestCaseConfig(invocations = 5, threads = 2)

   override fun defaultConfig() = TestCaseConfig(invocations = 5, threads = 2)
}
``` 
and result of it
```
Cannot execute multiple invocations in parent tests
java.lang.IllegalStateException: Cannot execute multiple invocations in parent tests
	at io.kotest.core.runtime.TestExecutor.invokeTestCase(TestExecutor.kt:146)
```
I think this situation is not obvious. It need to be described in docs or resolved in code

**Comment by sksamuel** [comment]
We should add a defaultInvocations option to Spec and Project Config.

**Comment by sksamuel** [comment]
Included as part of 4.6.0
